### PR TITLE
New package: cdb-0.75

### PR DESCRIPTION
**Pull request by liwakura**

### CODE REVIEW

**Comment by Vaelatern** [comment]
The problem is the compiler chosen is always `gcc -O2`. I'm preparing a patch to alter this behavior.

**Comment by Vaelatern** [comment]
Patch still missing parts, but...

```
--- Makefile.bck        2016-02-14 14:04:24.983231149 -0500
+++ Makefile    2016-02-14 14:24:09.764089757 -0500
@@ -167,7 +167,7 @@
 compile: \
 warn-auto.sh conf-cc
        ( cat warn-auto.sh; \
-       echo exec "`head -1 conf-cc`" '-c $${1+"$$@"}' \
+       echo exec "$(CC) $(CFLAGS)" '-c $${1+"$$@"}' \
        ) > compile
        chmod 755 compile
 
@@ -210,7 +210,7 @@
 warn-auto.sh conf-ld
        ( cat warn-auto.sh; \
        echo 'main="$$1"; shift'; \
-       echo exec "`head -1 conf-ld`" \
+       echo exec "$(CC) $(LDFLAGS)" \
        '-o "$$main" "$$main".o $${1+"$$@"}' \
        ) > load
        chmod 755 load
```

**Comment by Vaelatern** [comment]
Closer now. Still need to turn the install program into a simple shell script, so this is not yet enough.
To get this to work, `hostmakedepends="perl"` had to be added to the template.

Thanks to @chneukirchen for the perl.

```
--- Makefile.bck        2016-02-14 14:04:24.983231149 -0500
+++ Makefile    2016-02-14 14:24:09.764089757 -0500
@@ -12,17 +12,9 @@
 compile alloc.c alloc.h error.h
        ./compile alloc.c
 
-auto-str: \
-load auto-str.o buffer.a unix.a byte.a
-       ./load auto-str buffer.a unix.a byte.a 
-
-auto-str.o: \
-compile auto-str.c buffer.h readwrite.h exit.h
-       ./compile auto-str.c
-
 auto_home.c: \
-auto-str conf-home
-       ./auto-str auto_home `head -1 conf-home` > auto_home.c
+conf-home
+       perl -e 'printf "char %s[] = \"", shift; printf "\\%.3o", ord for shift =~ /./g; print "\";\n"' auto_home `head -1 conf-home` > auto_home.c
 
 auto_home.o: \
 compile auto_home.c
@@ -167,7 +167,7 @@
 compile: \
 warn-auto.sh conf-cc
        ( cat warn-auto.sh; \
-       echo exec "`head -1 conf-cc`" '-c $${1+"$$@"}' \
+       echo exec "$(CC) $(CFLAGS)" '-c $${1+"$$@"}' \
        ) > compile
        chmod 755 compile
 
@@ -210,7 +210,8 @@
 warn-auto.sh conf-ld
        ( cat warn-auto.sh; \
        echo 'main="$$1"; shift'; \
-       echo exec "`head -1 conf-ld`" \
+       echo exec "$(CC) $(CFLAGS) $(LDFLAGS)" \
        '-o "$$main" "$$main".o $${1+"$$@"}' \
+       $(LIBS) \
        ) > load
        chmod 755 load
```

File: srcpkgs/cdb/patches/errno.patch [hunk 1]
```diff
@@ -0,0 +1,9 @@
+*** error.h	2000-02-19 20:42:05.000000000 +0000
+--- error.h	2016-02-22 10:46:12.878627124 +0000
```
Comment (Vaelatern) [inline srcpkgs/cdb/patches/errno.patch]: Your process to make patches will probably be happiest if you use the `diff` command line command.
```
cp file file.bck
edit file
diff -u file.bck file >> /path/to/patch/file
```

Comment (Vaelatern) [inline srcpkgs/cdb/patches/errno.patch]: Remember to be in the root directory of the cdb extraction when you do it. In this case, if you can't see the Makefile in the same directory, you are in the wrong place for sure.

**Comment by Vaelatern** [comment]
At present I'm working on getting the install script written using simple shell instead of a whole C program.

**Comment by liwakura** [comment]
I duplicated the functionality of the ./install and hier.c in the Makefile. I'll pray to god to forgive my sins.

**Comment by liwakura** [comment]
Instead of massively patching the makefile, the installation is now done manually via the do_install hook.

File: srcpkgs/cdb/template [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'cdb'
+pkgname=cdb
+version=0.75
+revision=1
+build_style=gnu-makefile
+short_desc="CDB utilities"
+maintainer="Lain <lain@waifu.club>"
+license="Public domain"
+homepage="https://cr.yp.to"
```
Comment (Vaelatern) [inline srcpkgs/cdb/template]: Should be the homepage of the project, not the person

**Comment by Vaelatern** [comment]
Closed in d1efab1

**Pull request closed without merging**